src/daemon/http/mod.rs

```diff
@@ -357,7 +358,13 @@ impl Request {
     pub fn user_agent(&self) -> Option<String> {
         match self.headers().get(&USER_AGENT) {
             None => None,
-            Some(value) => value.to_str().ok().map(|s| s.to_string()),
+            Some(value) => value.to_str().ok().map(|s| {
+                if s.len() > HTTP_USER_AGENT_TRUNCATE {
+                    s[..HTTP_USER_AGENT_TRUNCATE].to_string()
```

Comment:
Is it safe to index into/slice a string, aren't there issues without byte boundaries being different to "character" boundaries?

Reply:
Yes, because of `HeaderValue::to_str`. I added a comment to make this more clear.